model/src/main/java/com/basistech/rosette/apimodel/DocumentRequest.java

```diff
@@ -81,13 +78,29 @@
      */
     private final O options;
 
+
+    public DocumentRequest(String profileId,
+                           LanguageCode language,
+                           Object content,
+                           String contentUri,
+                           String contentType,
+                           O options) {
+        super(profileId);
+        this.language = language;
+        this.content = content;
+        this.contentUri = contentUri;
+        this.contentType = contentType;
+        this.genre = null;
+        this.options = options;
+    }
+
     @Builder     // workaround for inheritance https://github.com/rzwitserloot/lombok/issues/853
     public DocumentRequest(String profileId,
```

Comment:
What's the best practice here?  Only mark the parameters as `@Deprecated` or should this whole method be marked as `@Deprecated`?